mesonbuild/backend/ninjabackend.py

```diff
@@ -3375,6 +3394,12 @@ def generate_link(self, target: build.BuildTarget, outname, obj_list, linker: T.
         else:
             dependencies = target.get_dependencies()
         internal = self.build_target_link_arguments(linker, dependencies)
+        #In AIX since shared libraries are archived the dependencies must
+        #depend on .a file with the .so and not directly on the .so file.
+        if mesonlib.is_aix ():
+            for i, val in enumerate(internal):
+                internal[i] = internal[i].replace ('.so', '.a')
+                internal[i] = re.sub ('[.][a]([.]?([0-9]+))*([.]?([a-z]+))*','.a', internal[i])
```

Comment:
Similar comments

Reply:
This is done. Kindly check the latest commit